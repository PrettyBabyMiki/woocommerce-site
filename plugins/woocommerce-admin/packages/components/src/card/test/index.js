/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import Card from '../';

describe( 'Card', () => {
	test( 'it renders correctly', () => {
		const { container, getByRole } = render(
			<Card title="A Card Example" />
		);
		expect( container ).toMatchSnapshot();

		// should have correct title
		expect(
			getByRole( 'heading', { name: 'A Card Example' } )
		).toBeInTheDocument();

		// should have correct class
		expect(
			container.getElementsByClassName( 'woocommerce-card' )
		).toHaveLength( 1 );
	} );
} );
