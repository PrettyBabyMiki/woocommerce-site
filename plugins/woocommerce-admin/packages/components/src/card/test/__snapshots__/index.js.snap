// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card it renders correctly 1`] = `
<div>
  <div
    class="woocommerce-card"
  >
    <div
      class="woocommerce-card__header"
    >
      <div
        class="woocommerce-card__title-wrapper"
      >
        <h2
          class="woocommerce-card__title woocommerce-card__header-item"
        >
          A Card Example
        </h2>
      </div>
    </div>
    <div
      class="woocommerce-card__body"
    />
  </div>
</div>
`;
